pagegraph-cli/src/adblock_rules.rs

```diff
@@ -3,8 +3,11 @@
 use pagegraph::graph::PageGraph;
 use pagegraph::types::{EdgeType, NodeType};
 
-pub fn main(graph: &PageGraph, filter_rule: &str) {
-    let matching_elements = graph.resources_matching_filter(&filter_rule);
+pub fn main(graph: &PageGraph, filter_rules: Vec<String>) {
+    let mut matching_elements = vec![];
+    for filter_rule in &filter_rules {
+        matching_elements.extend(graph.resources_matching_filter(&filter_rule));
+    }
```

Comment:
This won't actually work correctly for most real-world lists - for example, if you have the following list:

```
||example.com^
@@||example.com/index.html
```

Checking each item individually would mark a request to `https://example.com/index.html` as blocked, even though there is an exception for it.

You'd probably want to also modify the `resources_matching_filter` method to take in a list of filters, and apply them all at once.

Reply:
I added a new function https://github.com/brave-experiments/pagegraph-rust/pull/17/files#diff-b5cd7059084e703ee305e32cfb420f22412155f4080c5191c1c4f62fce796c25R436 that uses `Engine::from_rules` and then `check_network_urls_with_hostnames()`

Also added a test for this scenario in `sugarcoat-pipeline`

Reply:
(thanks again for catching this!)